soweego/importer/imdb_dump_extractor.py

```diff
@@ -0,0 +1,390 @@
+                    self.n_actors += 1
+                    types_of_entities.append(imdb_entity.ImdbActorEntity())
+
+                if "director" in professions:
+                    self.n_directors += 1
+                    types_of_entities.append(imdb_entity.ImdbDirectorEntity())
+
+                if "producer" in professions:
+                    self.n_producers += 1
+                    types_of_entities.append(imdb_entity.ImdbProducerEntity())
+
+                if any(prof in ["sound_department", "composer",
+                                "music_department", "soundtrack"]
+                       for prof in professions):
+                    self.n_musicians += 1
+                    types_of_entities.append(imdb_entity.ImdbMusicianEntity())
+
+                if "writer" in professions:
+                    self.n_writers += 1
+                    types_of_entities.append(imdb_entity.ImdbWriterEntity())
+
+                # if the only profession a person has is `miscellaneous` then we
+                # add it to all tables
+                if professions == "miscellaneous":
+                    self.n_misc += 1
+                    types_of_entities = [
+                        imdb_entity.ImdbActorEntity(),
+                        imdb_entity.ImdbDirectorEntity(),
+                        imdb_entity.ImdbMusicianEntity(),
+                        imdb_entity.ImdbProducerEntity(),
+                        imdb_entity.ImdbWriterEntity(),
+                    ]
+
+                # add person to every matching table
+                for etype in types_of_entities:
+                    self._populate_person(etype, person_info, session)
+
+                # if person is known for any movies then add these to the
+                # databse as well
+                if person_info.get("knownForTitles"):
+                    self.n_person_movie_links += 1
+                    self._populate_person_movie_relations(person_info, session)
+
+                # commit results to the database every `_sqlalchemy_commit_every` loops
+                if self.n_persons % self._sqlalchemy_commit_every == 0:
+                    session.commit()
+
+                self.n_persons += 1
+
+            # finally commit remaining entities
+            session.commit()
+
+        end = datetime.datetime.now()
+        LOGGER.info("Person import completed in %s. "
+                    "Total people imported: %d - "
+                    "Actors: %d - Directors: %d - Musicians: %d - "
+                    "Producers: %d - Writers: %d - Misc: %d",
+                    end - start, self.n_persons, self.n_actors,
+                    self.n_directors, self.n_musicians, self.n_producers,
+                    self.n_writers, self.n_misc)
+
+    def _populate_person(self, person_entity: imdb_entity.ImdbPersonEntity,
+                         person_info: Dict,
+                         session: object) -> None:
+        """
+        Given an instance of
```

Comment:
I really appreciate such accurate docstrings.
However, I believe they are not needed for private methods.
I think you can safely keep all those you wrote until now, but in the future just do that for public stuff.

Reply:
ok! I'll keep it in mind for future commits.